src/app/signup/signup.component.ts

```diff
@@ -3,46 +3,31 @@ import { Router } from '@angular/router';
 import { UserService } from '../user.service';
 import { FormGroup, FormBuilder, Validators } from "@angular/forms";
 
-import { MoviesComponent } from '../movies/movies.component';
-
 @Component({
   selector: 'abc-signup',
   templateUrl: './signup.component.html',
   styleUrls: ['./signup.component.scss']
 })
 export class SignupComponent implements OnInit {
 
-  signupForm: FormGroup;
-    id: '';
-    firstname: '';
-    lastname:'';
-    username:'';
-    password:'';
-    organisation: string='';
-    email: '';
-    contact: '';
-    
-    constructor(private router: Router, private userService: UserService, private formBuilder: FormBuilder) { }
+  form: FormGroup;
 
-    ngOnInit(): void {
-      this.signupForm = this.formBuilder.group({
-          firstname: [null, Validators.required],
-          lastname: [null, Validators.required],
-          username: [null, Validators.required],
-          password: [null, Validators.required],
-          organisation: [null, Validators.required],
-          email: [null, Validators.required],
-          contact:[null, Validators.required]
-      });
-    } 
+  constructor(private router: Router, private userService: UserService, private formBuilder: FormBuilder) { }
 
-  onSubmitNewUser(){
-    this.userService.addUser(this.signupForm.value)
-      .subscribe((res: any) => {
-        //const id = res._id;
-        this.router.navigate(['/movies']);
-      });
+  ngOnInit(): void {
+    this.form = this.formBuilder.group({
+      firstname: [null, [Validators.required]],
+      lastname: [null, [Validators.required]],
+      username: [null, [Validators.required]],
+      password: [null, [Validators.required]],
+      email: [null, [Validators.required]],
+      contact: [null]
+    });
   }
 
+  async signUp() {
+    await this.userService.register(this.form.value);
+    this.router.navigate(['/movies']);
+  }
```

Comment:
Again subscribe method not used. Why is this method preferred?

Reply:
That's because we changed `register` to return a `Promise` instead which is `awaitable`. Using promises instead of observables here makes sense since registering will likely emit only one event.